controllers/discovery/initDiscoveryUserData.js

```diff
@@ -107,6 +107,7 @@ const InitDiscoveryUserData = async (req, res) => {
           AND A.is_anonymous = false 
           AND A.is_banned = false
           AND A.karma_score > :minimumKarmaScore
+          AND A.user_id != :admin_user_id
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The change is logically correct as it excludes the admin user from the search results. However, it's important to note that this approach relies on the `BETTER_ADMIN_ID` environment variable being correctly set and maintained. If this value is not set or is changed without updating the environment variable, the functionality will not work as expected.

```diff
- AND A.user_id != :admin_user_id
+ AND A.user_id != (process.env.BETTER_ADMIN_ID || 'default_admin_id')
```
In the above code snippet, `'default_admin_id'` should be replaced with a valid default admin ID.



<!-- This is an auto-generated comment by OSS CodeRabbit -->